nifi-nar-bundles/nifi-iceberg-bundle/nifi-iceberg-processors/src/main/java/org/apache/nifi/processors/iceberg/converter/GenericDataConverters.java

```diff
@@ -47,45 +51,106 @@
  */
 public class GenericDataConverters {
 
-    static class SameTypeConverter extends DataConverter<Object, Object> {
+    static class PrimitiveTypeConverter extends DataConverter<Object, Object> {
+        final Type.PrimitiveType targetType;
+        final DataType sourceType;
+
+        public PrimitiveTypeConverter(final Type.PrimitiveType type, final DataType dataType) {
+            targetType = type;
+            sourceType = dataType;
+        }
 
         @Override
         public Object convert(Object data) {
-            return data;
+            switch (targetType.typeId()) {
+                case BOOLEAN:
+                    return DataTypeUtils.convertType(data, RecordFieldType.BOOLEAN.getDataType(), null);
```

Comment:
I think you can call directly the type conversion functions eg.: `DataTypeUtils.toBoolean`, `DataTypeUtils.toInteger` etc.